smtp_connector.py

```diff
@@ -63,106 +155,288 @@ def _validate_email(self, input_data):
+            return phantom.APP_ERROR, "Client ID has been changed. Please run Test Connectivity again."
+
+        request_url = config.get("token_url")
+
+        body = {
+            'grant_type': 'refresh_token',
+            'client_id': client_id,
+            'refresh_token': self._refresh_token,
+            'client_secret': client_secret
+        }
 
-        # UnicodeDammit(msg.as_string()).unicode_markup.encode("utf-8")
-        # Above fix works for both Python v2 and Python v3 and should not be changed or
-        # changed only after thorough testing on both the Python v2 and v3 versions.
-        # This fix is based on solutions provided on the Python bug tracker portal issue
-        # raised by us for sendemail not working for Unicode Characters for Python v3
-        # Bug Link - https://bugs.python.org/issue41023
         try:
-            if self._python_version != 2:
-                root_data_str = UnicodeDammit(root_data_str).unicode_markup.encode('utf-8')
-        except:
-            self.debug_print("Error occurred while handling python 2to3 compatibility for the email message string")
+            r = requests.post(request_url, data=body, timeout=DEFAULT_REQUEST_TIMEOUT)
+        except Exception as e:
+            return phantom.APP_ERROR, "Error refreshing token: {}".format(str(e))
 
-        return root_data_str
+        try:
+            response_json = r.json()
+            if response_json.get("error"):
+                return phantom.APP_ERROR, "Invalid refresh token. Please run the test connectivity again."
+            oauth_token.update(r.json())
+        except Exception:
+            return phantom.APP_ERROR, "Error retrieving OAuth Token"
+
+        self._access_token = oauth_token.get('access_token')
+
+        self._state['oauth_token'] = oauth_token
+        return phantom.APP_SUCCESS, ""
+
+    def _set_interactive_auth(self, action_result):
+
+        config = self.get_config()
+        client_id = config.get("client_id")
+        client_secret = config.get("client_secret")
+
+        # Run the initial authentication flow only if current action is test connectivity
+        if self.get_action_identifier() != phantom.ACTION_ID_TEST_ASSET_CONNECTIVITY:
+            if not self._access_token:
+                return phantom.APP_ERROR, "Unable to get access token. Has Test Connectivity been run?"
+
+            try:
+                ret_val = self._connect_to_server(action_result)
+            except Exception as e:
+                return self._parse_connection_error(action_result, e), action_result.get_message()
+
+            if phantom.is_fail(ret_val):
+                return phantom.APP_ERROR, action_result.get_message()
+
+            self.save_state(self._state)
+        else:
+            self.debug_print("Try to generate token from authorization code")
+            asset_id = self.get_asset_id()
+            rsh = RequestStateHandler(asset_id)  # Use the states from the OAuth login
```

Comment:
Instead of debug_print() use error_print() method to print the exception message as error.